l2geth/core/vm/ovm_state_manager.go

```diff
@@ -156,6 +158,32 @@ func putContractStorage(evm *EVM, contract *Contract, args map[string]interface{
 	return []interface{}{}, nil
 }
 
+func putAccountCode(evm *EVM, contract *Contract, args map[string]interface{}) ([]interface{}, error) {
+	address, ok := args["_address"].(common.Address)
+	if !ok {
+		return nil, errors.New("Could not parse address arg in putAccountCode")
+	}
+	code, ok := args["_code"].([]byte)
+	if !ok {
+		return nil, errors.New("Could not parse code arg in putAccountCode")
+	}
+
+	// save the block number and address with modified key if it's not an eth_call
+	if evm.Context.EthCallSender == nil {
```

Comment:
Note that during `debug_traceTransaction` the `EthCallSender` is `nil` and will hit this block. We don't have a good solution to this right now

Reply:
What is the problem you are referring to a solution of here?  I added this because the diff proof should include changed accounts if we want to prove fraud on an upgrade tx.